(GH-2755) Add support for running plans in noop mode (#2822)
puppetlabs/bolt: documentation/testing_plans.md

end
```

### Running plans in no-operation mode

> The `run_plan` function has a `noop` option which can be used to run a plan in
> no-operation (no-op) mode. To run a plan in no-op mode, set the `noop` option
> to `true`:

Reviewer: ```suggestion
Use the `noop` option on the `run_plan` function to run a plan in
no-operation (no-op) mode. To run a plan in no-op mode, set the `noop` option
to `true`:
```

```ruby
run_plan('myplan', {}, noop: true)
```

When you run a plan in no-op mode, all tasks, apply blocks, and sub-plans are
automatically run in no-op mode. If a plan run in no-op mode invokes any of

> the following functions, then the plan automatically fails:

Reviewer: ```suggestion
the following functions, the plan automatically fails:
```

- `download_file`
- `file::write`
- `run_command`
- `run_script`
- `upload_file`
- `write_file`

Because Bolt automatically fails when these functions are called in no-op mode,
you can write tests that ensure the action is never invoked using
[mocks](#stubs-and-mocks). For example, the following plan invokes the
`run_command` function:

```puppet
plan command (
  TargetSpec $targets = 'localhost'
) {
  $result = run_command('touch example.txt', $targets)
}
```

> You can write a test that ensures that the `run_command` function is not
> invoked and that the plain fails with a helpful message:

Reviewer: ```suggestion
You can write a test that ensures that the plan does not invoke
`run_command` and that the plan fails with a helpful message:
```
